src/circuit/gadget/ecc/chip.rs

```diff
@@ -276,12 +280,6 @@ impl EccChip {
             config.create_gate(meta);
         }
 
-        // Create incomplete point addition gate
-        {
-            let config: add_incomplete::Config = (&config).into();
-            config.create_gate(meta);
-        }
```

Comment:
The change to `circuit_description` is more complex than it needs to be, because `add_incomplete::Config` is refactored before `witness_point::Config`. This causes constraints to be reordererd in the first commit, that are subsequently un-reordered in the third commit when `witness_point::Config` is refactored and one more configured before `add_incomplete::Config`.

If the commits were reordered so that they refactored these gates in the order they are configured in the circuit, then the only change to `circuit_description` should be renumbering fixed columns (or replacing standalone selector columns with compressed selector columns, or vice versa), due to moving the sub-configs above the main config's selector creation. I manually confirmed this to be the case for the first three commits (in fact, as a result the `add_incomplete::Config` and `add::Config` refactors cause no circuit change).

Reply:
[Force-pushed](https://github.com/zcash/orchard/compare/8d433ff62d2bba5ab8c4a51b2aa62c5737a7cb8a..5671355ade4008a052 750cad8247998743dad266) to reorder the commits as suggested.